lib/puppet/network/http/connection.rb

```diff
@@ -208,12 +208,22 @@ def apply_options_to(request, options)
     def execute_request(connection, request)
       start = Time.now
       connection.request(request)
-    rescue EOFError => e
+    rescue Exception => exception
       elapsed = (Time.now - start).to_f.round(3)
       uri = @site.addr + request.path.split('?')[0]
-      eof = EOFError.new(_('request %{uri} interrupted after %{elapsed} seconds') % {uri: uri, elapsed: elapsed})
-      eof.set_backtrace(e.backtrace) unless e.backtrace.empty?
-      raise eof
+      eclass = exception.class
+
+      err = case exception
```

Comment:
Looks like a valid test failure with TravisCI. Also two small issues:

It's bad to rescue `Exception` as it includes all exceptions like `NoMemoryError` and `SystemExit`. Best to just `rescue => exception` which will rescue `StandardError` and its subclasses. See https://blog.honeybadger.io/understanding-the-ruby-exception-hierarchy/.

The case/when statement uses `===` (aka triple equals) for comparison so [you can drop the `exception.class` bit](http://batsov.com/articles/2012/10/14/ruby-tip-number-3-matching-on-an-objects-class-in-a-case-expression/) and just do:

```ruby
case exception
when EOFError
  ...
```

But don't do:

```ruby
case exception.class
when EOFError
  ...
```

Also I recently found a bug in ruby that I think explains why [agents sometimes hang indefinitely or timeout](https://bugs.ruby-lang.org/issues/14972).

Reply:
Changed to `rescue exception`. Re: `exception.class`, that's not being used in the case statement comparison, it's used when building the new exception to raise. I believe the case statement matches your recommendation; am I missing something?

That bug sounds like it could totally be impacting us.

Reply:
Sorry, completely misread the case statement, doh!